### PR TITLE
feat: oidc implementation 

### PR DESCRIPTION
**Pull request by brivu**
This `PR` makes use of CircleCI's `OIDC Token` to generate temporary `AWS` keys, enabling users to run `ecs` jobs  more securely without having to store static keys as environment variables. 

To generate temporary tokens, provide the job with a `role-arn` and `role-session-name`.

### CODE REVIEW

File: src/jobs/deploy-service-update.yml [hunk 1]
```diff
@@ -21,11 +21,25 @@ parameters:
...
     type: string
     default: ''
+  role-arn:
+    description: |
+      The Amazon Resource Name (ARN) of the role that the caller is assuming.
+      Role ARN must be configured for web identity.
+    type: string
+    default: ""
+  role-session-name:
+    description: An identifier for the assumed role session
```
Comment (Jaryt) [inline src/jobs/deploy-service-update.yml:36]: Will this be evaluated in the bash script? Might be good to note that here if so.

**Review by Jaryt — COMMENTED** [review]
Looking good, just those two description changes needed

File: src/commands/update-service.yml [hunk 2]
```diff
@@ -71,17 +71,26 @@ parameters:
     default: ''
   container-secret-updates:
     description: >
-      Use this to update or set the values of secrets variables that will be
-      defined for the containers.
-      (Existing secrets variables not included in this parameter will not be
-      removed)
+      Use this to update or set the values of secret variables that will be defined for the containers.
+      (Existing secrets variables not included in this parameter will not beremoved)
```
[the remark this answers is not in the document]
Comment (brivu) [reply src/commands/update-service.yml:75]: good catch

**Review by Jaryt — APPROVED** [review]
lgtm!

**Pull request merged**